content/java/concepts/treemap/treemap.md

```diff
@@ -12,32 +12,37 @@ CatalogContent:
   - 'paths/computer-science'
 ---
 
-A `TreeMap` uses the [`Map`](https://www.codecademy.com/resources/docs/java/Map) interface, which means it stores items as key-value pairs.
+A **`TreeMap`** uses the [`Map`](https://www.codecademy.com/resources/docs/java/map) interface, which means it stores items as key-value pairs.
 
-They are like the [`HashMap`](https://www.codecademy.com/resources/docs/java/hashmap). However, a `HashMap`, the `TreeMap` stores its items sorted by the natural ordering of its keys. This is due to the differing means of storage each type of collection uses internally. The `HashMap` stores its keys as hashes for lookup, whereas the `TreeMap` stores its keys in a binary tree structure for lookup. The tradeoff for this natural ordering of the `TreeMap` is that its operations are slower than a `HashMap`.
+Similiar to a [`HashMap`](https://www.codecademy.com/resources/docs/java/hashmap), the `TreeMap` stores key-value pairs. The difference is the means of storage each type of collection uses internally. A `HashMap` stores its keys as hashes in a [`Hash Table`](https://www.codecademy.com/resources/docs/general/hash-table) whereas a `TreeMap` stores its keys in a ['Binary Tree Structure'](https://www.codecademy.com/resources/docs/general/binary-search-tree), which is a type of tree data structure that has at most two children nodes per parent.
+
+While the structure of a 'TreeMap' is more orderly, it comes as a tradeoff with speed in comparison to a 'HashMap'. In [Big-O notation](https://www.codecademy.com/resources/docs/general/big-o-notation), most `HashMap` operations run in amortized O(1) time; they are generally more time-efficient.  Most `TreeMap` operations run in O(log(n)) time and the `TreeMap` is generally more space-efficient due to its dynamic allocation of memory per value added.
+
+> **Note:** Complexity is a crucial factor when determining what structure to implement. If you want faster operations over your data, you would want to aim for a `HashMap`. However, if you're looking to keep your values sorted, it's best to stick to a `TreeMap`.
```

Comment:
Is there any way we can rephrase this note so that "you"/"your" is not included in the text? For Docs entries, we prefer that entries have a more neutral, manual tone that just presents information.

Reply:
@Dusch4593 Good thought, let me change the you/your